### PR TITLE
Feat: Add `monitor_metric` config option for Noise2Void

### PR DESCRIPTION
**Pull request by Callidior**
## Disclaimer

- [ ] I am an AI agent.
- [x] I have used AI and I thoroughly reviewed every line.
- [ ] I have not used AI extensively.

## Description

> [!NOTE]  
> Add `monitor_metric` config option to customize the metric used by `ReduceLROnPlateau` LR scheduler.

### Background - why do we need this PR?

Enables training without validation runs.

### Overview - what changed?

- Added option `monitor_metric` to factory functions

## How has this been tested?

- Unit tests still pass
- Training runs on own data still successful and yielding expected results, both with monitoring `val_loss` and monitoring `train_loss_epoch`

## Additional Notes and Examples

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [ ] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

on-behalf-of: @ZEISS <bjoern.barz@zeiss.com>

### CODE REVIEW

File: src/careamics/config/algorithms/unet_algorithm_config.py [hunk 1]
```diff
@@ -69,6 +69,9 @@ class UNetBasedAlgorithm(BaseModel):
     lr_scheduler: LrSchedulerConfig = LrSchedulerConfig()
     """Learning rate scheduler to use, defined in SupportedLrScheduler."""
 
+    monitor_metric: str = "val_loss"
```
Comment (jdeschamps) [inline src/careamics/config/algorithms/unet_algorithm_config.py]: I think this should move to `N2VAlgorithm` child class since we don't expect it to be used with CARE. In addition, I'd make it of type `Literal["train_loss", "val_loss"]` for clarity.

File: src/careamics/config/ng_factories/care_n2n_factory.py
Comment (jdeschamps) [inline src/careamics/config/ng_factories/care_n2n_factory.py:1]: In line with the previous comment, I'd not add this parameter to the CARE/N2N convenience functions.

File: src/careamics/lightning/dataset_ng/lightning_modules/care_module.py
Comment (jdeschamps) [inline src/careamics/lightning/dataset_ng/lightning_modules/care_module.py:1]: In line with the previous comments, I'd remove it from here.

**Review by jdeschamps — CHANGES_REQUESTED** [review]
Thanks @Callidior! Sorry it took so long to review, it was a busy beginning of the week.

I would restrict the monitoring of `train_loss` to Noise2Void, since it would only be used there.

File: src/careamics/config/algorithms/unet_algorithm_config.py (hunk 1 above)
Comment (Callidior) [reply src/careamics/config/algorithms/unet_algorithm_config.py]: I'd rather use `"train_loss_epoch"` instead of the noisy step-wise loss. Is it okay to add that as option as well?

Comment (jdeschamps) [reply src/careamics/config/algorithms/unet_algorithm_config.py]: Absolutely, didn't think of that!

**Review by jdeschamps — APPROVED** [review]

**Comment by jdeschamps** [comment]
Thanks!
